### PR TITLE
reduce copy pasta overhead on all applications

### PR DESCRIPTION
**Pull request by svenwltr**
Reduce boilerplate in our code. Usage example is in https://github.com/rebuy-de/golang-template/pull/12.

### CODE REVIEW

**Comment by svenwltr** [comment]
@rebuy-de/prp-rebuy-go-sdk Please review. Is the documentation understandable?

File: cmdutil/doc.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+//     defer cmdutil.HandleExit()
+//     run()
+//   }
+//
+//   func run() {
+//     defer fmt.Println("important cleanup")
+//     err := doSomething()
+//     if err != nil {
+//       log.Error(err)
+//       cmtutil.Exit(2)
```
Comment (bjoernhaeuser) [inline cmdutil/doc.go]: cmd :)

Comment (svenwltr) [reply cmdutil/doc.go]: fixed

File: cmdutil/doc.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+// providing basic application behavior and for reducing boilerplate code.
+//
+// An example application can be found at
+// https://github.com/rebuy-de/golang-template.
+//
+// Graceful Application Exits
+//
+// In many command line applications it is desired to exit the process
+// immediately, if it is clear that the application cannot recover. Important
+// node: This is designed for actual applications (ie not libraries), because
```
Comment (bethge) [inline cmdutil/doc.go]: _note_

File: cmdutil/doc.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+//           `Your name.`)
+//   }
+//
+//   func NewRootCommand() *cobra.Command {
+//       cmd := cmdutil.NewRootCommand(new(App))
+//       cmd.Short = "an example app for golang which can be used as template"
+//       return cmd
+//   }
+//
+// The App struct contains fields for parameters which are defined Bind or for
```
Comment (bethge) [inline cmdutil/doc.go]: >... which are defined Bind ...

which are defined by Bind ... ?

File: cmdutil/doc.go (hunk 2 above)
Comment (svenwltr) [reply cmdutil/doc.go]: fixed

File: cmdutil/doc.go (hunk 3 above)
Comment (svenwltr) [reply cmdutil/doc.go]: I changed it to "in Bind". Does this make sense?

File: cmdutil/doc.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+// only the application itself should decide when to exit. Libraries should
+// alway return errors.
+//
+// There are three ways to handle fatal errors in Go. With os.Exit() the
+// process will terminate immediately, but it will not call any deferrers which
+// means that possible cleanup task do not get called. The next way is to call
+// panic, which respects the defer statements, but unfortunately it is not
+// possible to define an exit code and the user gets confused with a stack
+// trace. Finally, the function could just return an error indicating that
+// things failed, but this intriduces a lot of code and conditionals and
```
Comment (stephanlindauer) [inline cmdutil/doc.go]: *introduces

Comment (stephanlindauer) [reply cmdutil/doc.go]: [...] lot of code, conditionals and [...]

File: cmdutil/doc.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+// appears uncecessary, when it is already clear that the application cannot
```
Comment (stephanlindauer) [inline cmdutil/doc.go]: unnecessary

File: cmdutil/doc.go (hunk 4 above)
Comment (svenwltr) [reply cmdutil/doc.go]: fixed

File: cmdutil/doc.go (hunk 5 above)
Comment (svenwltr) [reply cmdutil/doc.go]: fixed

**Comment by svenwltr** [comment]
* https://godoc.org/github.com/rebuy-de/rebuy-go-sdk/cmdutil
* https://github.com/rebuy-de/rebuy-go-sdk/releases/tag/v1.1.0

**Pull request merged**